File: library/src/main/java/me/proxer/library/entity/list/TranslatorGroupCore.java
package me.proxer.library.entity.list;

import com.squareup.moshi.Json;
import lombok.Getter;
import lombok.Value;
import me.proxer.library.entity.ProxerIdItem;
import me.proxer.library.entity.ProxerImageItem;
import me.proxer.library.enums.Country;

/**
 * Entity containing the core information of a translator group.
 *
 * @author Ruben Gees
 */
@Value
public class TranslatorGroupCore implements ProxerIdItem, ProxerImageItem {

    /**
     * Returns the id.
     */
    @Getter(onMethod = @__({@Override}))
    @Json(name = "id")
    private String id;

    /**
     * Returns the name.
     */
    @Json(name = "name")
    private String name;

    /**
     * Returns the country, the translator group is active in.
     */
    @Json(name = "country")
    private Country country;

    /**
     * {@inheritDoc}
     */
    @Getter(onMethod = @__({@Override}))
    @Json(name = "image")
    private String image;
}

File: library/src/main/kotlin/me/proxer/library/entity/list/TranslatorGroupCore.kt
package me.proxer.library.entity.list

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import me.proxer.library.entity.ProxerIdItem
import me.proxer.library.entity.ProxerImageItem
import me.proxer.library.enums.Country

/**
 * Entity containing the core information of a translator group.
 *
 * @property name The name.
 * @property country The country, the translator group is active in.
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = true)
data class TranslatorGroupCore(
    @Json(name = "id") override val id: String,
    @Json(name = "name") val name: String,
    @Json(name = "country") val country: Country,
    @Json(name = "image") override val image: String
) : ProxerIdItem, ProxerImageItem
